### DISCUSSION TITLE
posts/valueconverter-tips-and-tricks/

### DISCUSSION THREAD

**Discussion by giscus** [BYJRK/byjrk.github.io#24] [Announcements]
# posts/valueconverter-tips-and-tricks/

在这篇文章中，我们会介绍 WPF 的值转换器的一些实用技巧，相信一定可以帮助大家更好地使用值转换器，提高开发效率。

https://blog.coldwind.top/posts/valueconverter-tips-and-tricks/

**Comment by ProjektMing** [comment]
请问如果我需要给 RelayCommand 传入多个参数是不是一定要这样用转换器做成元组？

**Reply by BYJRK** [reply]
用MultiBinding，配合MultiValueConverter组合为数组再传进去

**Comment by carletery** [comment]
用过了 mvvm 之后，对 valueconverter 是否是合理设计有点疑问。
在 viewmodel 里面，通过在属性 get 里面进行针对其他属性的计算或转换，然后把该属性绑定到 xaml。感觉这样会比单独维护 valueconverter 要合理。一是转换逻辑贴近原始数据属性；二是无需在 valueconverter 中做类型转换；三是 xaml 本来就应该聚焦界面，带上数据转换属于非必要的脑力开销，而 viewmodel 做这些转换的事似乎更合理方便。
当然我现在只是简单的用 valueconverter，没有特别深入，不了解 valueconverter 是否有其他高级特性。

**Reply by BYJRK** [reply]
MVVM这东西我只能说，don't be so serious。很多时候你回过头来会发现，曾经纠结的一些东西都没有什么意义。只要别显著提高你后续维护及扩展的成本，那么通常都不会是什么问题。比如值转换器与VM中属性通知的取舍，是否应当将EventArgs传递给VM，在VM中使用MessageBox、OpenFileDialog是否合适，在VM中出现Brush、Visibility、FontFamily是否合适，等等

**Reply by carletery** [reply]
那就是说 valueconverter 并没有其他特殊的功能。那我当然选择 viewmodel 属性这样符合口味的了 :)
